hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/impl/ChangeDetectionPolicy.java

```diff
@@ -362,14 +363,14 @@ public void applyRevisionConstraint(CopyObjectRequest request,
         String revisionId) {
       if (revisionId != null) {
         LOG.debug("Restricting copy request to etag {}", revisionId);
-        request.withMatchingETagConstraint(revisionId);
+        request.toBuilder().copySourceIfMatch(revisionId);
```

Comment:
I suspect that the call to `toBuilder()` creates a new instance rather than modify the original `request`.